### ISSUE TITLE
10.8 AGS & Portal install error

### ISSUE THREAD

**Issue by minagim** [Esri/arcgis-powershell-dsc#276]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request

### Module Version

* 3.0.2

### Affected Resource(s)

* ArcGIS_Server
* ArcGIS_Portal

### Configuration Files

[1 BaseDeployment_STG_DataStore_Server_Portal_WA_DNS_MK_Github.json.txt](https://github.com/Esri/arcgis-powershell-dsc/files/4826683/1.BaseDeployment_STG_DataStore_Server_Portal_WA_DNS_MK_Github.json.txt)

### Expected Behavior

ArcGIS Servers & Portal are configured and federated with IWA.

### Actual Behavior

4 ArcGIS Servers & a Portal configured and federated

![Error - Duplicate Node Names](https://user-images.githubusercontent.com/60987985/85592650-5a5f0800-b614-11ea-903d-c6927159acb8.JPG)

### Steps to Reproduce

Invoke-ArcGISConfiguration -ConfigurationParametersFile D:\Scripts\script.json -Mode InstallLicenseConfigure

### Important Factoids

Hi,

I am trying to install 10.8 data store, 4 servers & a portal with federation on 8 VMs with Windows Server 2019 but getting the following error.

![Error - Duplicate Node Names](https://user-images.githubusercontent.com/60987985/85592688-61861600-b614-11ea-8679-1b4c30448744.JPG)

I am trying to install 4 different web adapters for 4 AGSs on one web server.
How do you configure a system with multiple AGSs?  Is it possible?
Any advice would be appreciated.

Mina

**Comment by cameronkroeker** [comment]
Hello @minagim 

To achieve this you will need to generate separate json files (4 total). One that installs the web adaptor and server, and then a second one that contains an identical copy of the server information but has new information for the second web adaptor node name, ServerContext, and AdminAccess. Then repeat this for the third and fourth web adaptors.

Thanks,
Cameron K.

**Comment by minagim** [comment]
@cameronkroeker Thanks for your prompt response.  After reading [#134](https://github.com/Esri/arcgis-powershell-dsc/pull/134), I thought it would be possible.  Thanks for confirming it.  I will close the case.

**Comment by minagim** [comment]
I am getting a couple of new errors while running the modified script.

First error is about the server license.

![image](https://user-images.githubusercontent.com/60987985/85742397-344b6d80-b6d1-11ea-926f-29d45d6c0e8b.png)

Second error is about the the portal default role.  

PowerShell DSC resource ArcGIS_Portal  failed to execute Set-TargetResource functionality with error message: [Error] - Set-PortalUserDefaults Response:- Invalid values for role and/or user license type parameters.  
The SendConfigurationApply function did not succeed.

![image](https://user-images.githubusercontent.com/60987985/85741208-39f48380-b6d0-11ea-94db-3bddb90be130.png)

I have the following parameters set in the script.

"EnableAutomaticAccountCreation": "true",
"DefaultRoleForUser": "account_user",
"DefaultUserLicenseTypeIdForUser": "viewerUT",

**Comment by cameronkroeker** [comment]
> I am getting a couple of new errors while running the modified script.
> 
> First error is about the server license.
> 
> ![image](https://user-images.githubusercontent.com/60987985/85742397-344b6d80-b6d1-11ea-926f-29d45d6c0e8b.png)
> 
> Second error is about the the portal default role.
> 
> PowerShell DSC resource ArcGIS_Portal failed to execute Set-TargetResource functionality with error message: [Error] - Set-PortalUserDefaults Response:- Invalid values for role and/or user license type parameters.
> The SendConfigurationApply function did not succeed.
> 
> ![image](https://user-images.githubusercontent.com/60987985/85741208-39f48380-b6d0-11ea-94db-3bddb90be130.png)
> 
> I have the following parameters set in the script.
> 
> "EnableAutomaticAccountCreation": "true",
> "DefaultRoleForUser": "account_user",
> "DefaultUserLicenseTypeIdForUser": "viewerUT",

Hello @minagim ,

The first error is indicating that the ArcGIS Server Authorization Wizard is returning an error when attempting to authorize the license. I recommend checking the license file to ensure it is valid (correct version, contains all necessary info like email, address, etc), and even perhaps attempt to manually authorize to see if it gives the same error.

The second error, try modifying `"EnableAutomaticAccountCreation": "true",` to `"EnableAutomaticAccountCreation": true,`. Boolean (true|false) should be all lowercase and not contain quotes.

**Comment by minagim** [comment]
Hi, @cameronkroeker,

Thanks for your answers.  I just tried to validate the license manually on the server and it worked.  What could it the issue?

![image](https://user-images.githubusercontent.com/60987985/85758300-30721800-b6de-11ea-8acf-5c985df49fcd.png)


Also, for the "EnableAutomaticAccountCreation" parameter, the double quotes around the boolean values in the sample script threw me off.  I am new to Powershell and have a lot to learn.
I made the change and still got the same error.

![image](https://user-images.githubusercontent.com/60987985/85758000-f30d8a80-b6dd-11ea-8ceb-8c74b54ea067.png)

![image](https://user-images.githubusercontent.com/60987985/85757900-da04d980-b6dd-11ea-9ffe-cbccb4e13a49.png)

**Comment by cameronkroeker** [comment]
Hi @minagim ,

Hm I'm not quite sure what could be wrong with the ArcGIS Server authorization. Two other suggestions I have are, 1. Try moving the authorization file to another location, 2. Try to manually authorize it silently, as this is how the PowerShell Module does it:

https://enterprise.arcgis.com/en/server/latest/install/windows/authorize-arcgis-server.htm#ESRI_SECTION1_2EF08A7261D248F596B749D3ED825460

For the second issue, try a combination of things. For example:
```
"DefaultRoleForUser": "org_user",
"DefaultUserLicenseTypeIdForUser": "viewerUT",
```

or
```
"DefaultRoleForUser": "account_user",
"DefaultUserLicenseTypeIdForUser": "editorUT",
```
I suspect the issue here is these user types must be compatible with the defaultRoleForUser that has been specified.

Values: fieldWorkerUT | GISProfessionalStdUT | GISProfessionalBasicUT | GISProfessionalAdvUT | creatorUT | editorUT | insightsAnalystUT

https://developers.arcgis.com/rest/enterprise-administration/portal/update-security-configuration.htm

https://developers.arcgis.com/rest/users-groups-and-items/set-user-default-settings.htm

**Comment by cameronkroeker** [comment]
Hello @minagim were you able to make any progress on this issue?

**Comment by minagim** [comment]
Hi, Cameron.

I am so sorry for the delay in reply.   I wasn’t able to make it work for both issues.

For the first issue, I moved the location around to see if it had any affect but couldn’t make it work. I am planning to test it a little more for PROD environment as I am done with our base deployment in STG.

For the second issue, I was wondering if it was a version-related issue as I was installing 10.8.  I tried different combinations but nothing worked.

I downloaded 10.8.1 versions this week and am excited  to test upgrading to 10.8.1 sometime next week.
I will go ahead and upload my answers for the open case and close it.

Thanks for all your help.

Mina
